spec/index.bs

```diff
@@ -4723,6 +4723,20 @@ must be well nested.
 
             **Returns:** {{undefined}}
 
+            If any of the following conditions are unsatisfied, generate a {{GPUValidationError}} and stop.
+            <div class=validusage>
+                - |this|.{{GPUCommandEncoder/[[state]]}} is {{encoder state/open}}.
+                - |querySet| is [$valid to use with$] |this|.
+                - |destination| is [$valid to use with$] |this|.
+                - |destination|.{{GPUBuffer/[[usage]]}} contains {{GPUBufferUsage/QUERY_RESOLVE}}.
+                - |firstQuery| is less than the number of queries in |querySet|.
+                - (|firstQuery| + |queryCount|) is less than or equal to the number of queries in |querySet|.
+                - The queries between [|firstQuery|, |firstQuery| + |queryCount| - 1] are written by query operations.
```

Comment:
I'm not sure about this one, an alternative is saying that all queries start at 0, so resolving unused queries writes 0 in the buffer.

Reply:
Vulkan validation layers will throw error if the queries for resolving contain a unused query. So if resolving unused queries to 0 in the buffer, we may need to sumulate by calculating the range of the used queries, then calling resolve method for many times if users skip some indexes when writing values.
For example:
```
encoder.writeTimestamp(querySet, 0);
encoder.writeTimestamp(querySet, 2);
encoder.writeTimestamp(querySet, 3);
encoder.writeTimestamp(querySet, 5);
```

In Vulkan implementaion, we need call vkCmdCopyQueryPoolResults three times:
```
vkCmdCopyQueryPoolResults(cmd, querySet, 0 /*firstIndex*/, 1 /*queryCount*/, dstBuffer, ....);
vkCmdCopyQueryPoolResults(cmd, querySet, 2 /*firstIndex*/, 2 /*queryCount*/, dstBuffer, ....);
vkCmdCopyQueryPoolResults(cmd, querySet, 5 /*firstIndex*/, 1 /*queryCount*/, dstBuffer, ....);
```
The buffer values for other query indexes, we can initialize them to 0, but cannot be worte via vkCmdCopyQueryPoolResults.

Reply:
Open it as a question in #1072 for more discussion, I will remove this one to land other validations.